src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java

```diff
@@ -342,7 +342,13 @@ public Long getFsGroup() {
 		}
 	}
 
-	public static class InitContainer {
+	public static class InitContainer extends ContainerProperties {
+	}
+
+	static class Container extends io.fabric8.kubernetes.api.model.Container {
```

Comment:
I am curious if we can use the same approach of using fabric8 api models to other supported properties as well. I believe this can eliminate some explicit models with a fewer properties (for instance the ContainerProperties we have for InitContainer). 

Also, I thought having an explicit type named `Container` could help here. But, I am ok if we can directly use the fabric8 Container type as is in this class.

Reply:
you could use the fabric8 objects, but thats a bit leakly. theres an open issue around this: https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/issues/296